Sources/SwiftInspectorCommands/ImportsCommand.swift

```diff
@@ -41,13 +42,14 @@ final class ImportsCommand: ParsableCommand {
   /// Runs the command
   func run() throws {
     let cachedSyntaxTree = CachedSyntaxTree()
-    let analyzer = ImportsAnalyzer(cachedSyntaxTree: cachedSyntaxTree)
+    let analyzer = StandardAnalyzer(cachedSyntaxTree: cachedSyntaxTree)
     let fileURL = URL(fileURLWithPath: path)
 
     let outputArray = try FileManager.default.swiftFiles(at: fileURL)
       .reduce(Set<String>()) { result, url in
-        let importStatements = try analyzer.analyze(fileURL: url)
-        let output = importStatements.map { outputString(from: $0) }
+        let reader = ImportSyntaxReader()
+        try analyzer.analyze(fileURL: url, withVisitor: reader)
+        let output = reader.imports.map { outputString(from: $0) }
```

Comment:
I don't really love this pattern either, I think it's weird that we need to pass the reader to another class and then _read from the reader_ rather than the Analyzer outputting this information directly.

Reply:
f6fad8b51e4f0ff3fea59cbdc13f728e1eab1231